### DISCUSSION TITLE
preUpgradeTasks

### DISCUSSION THREAD

**Discussion by Leksat** [renovatebot/renovate#8984] [Request Help] [answered]
Hello!

First of all, big thanks for the great tool! ❤️

We use NOT self-hosted Renovate. So far it works great. But we need to deal with Drupal updates.

The thing is that the Drupal configuration is stored in Git, but Drupal/module updates can modify it. I know that this maybe can be achieved with a self-hosted configuration using `postUpgradeTasks`. But we need one pre-upgrade task as well:
- Pre: Install Drupal using existing configuration.
- Update dependencies.
- Post: Run Drupal updates and export the updated config.

I'm not familiar with the self-hosted solution. Question: Is it possible to run pre-update tasks on a self-hosted configuration?

**Comment by viceice** [comment, marked as answer]
No `preUpgradeTasks` yet, follow #8804
